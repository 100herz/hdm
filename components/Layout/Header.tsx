import Link from "next/link"
import styled from "styled-components"

const Section = styled.section`
  position: relative;

  &::before {
    position: absolute;
    top: 0;
    left: 0;
    width: calc(2.9333 * 90%);
    height: 90%;
    background: url(/ci/hdm-bg.svg) no-repeat;
    opacity: 0.05;
    content: "";
  }

  @media (min-width: 768px) {
    position: static;

    &::before {
      left: 6rem;
      width: calc(2.9333 * 100%);
      height: 100%;
    }
  }
`

type Props = {
  className?: string
}

const Header: React.FC<Props> = ({ className = "" }: Props) => {
  return (
    <header className={`bg-gray-100 relative overflow-x-hidden ${className}`}>
      <div className="container justify-between my-6 lg:flex">
        <h1 className="pt-8">
          <img className="w-56 mx-auto" src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" />
          <span className="sr-only">HDM - Holz Design Moers GmbH</span>
        </h1>

        <nav className="flex justify-between my-16 font-semibold lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end">
          <Link href="/#produkte">
            <a>Produkte</a>
          </Link>
          <Link href="/#unternehmen">
            <a>Unternehmen</a>
          </Link>
          <Link href="/#kontakt">
            <a>Kontakt</a>
          </Link>
        </nav>
      </div>

      <Section className="container pt-12 pb-20 text-xl leading-10 lg:text-3xl lg:py-40">
        <p className="lg:max-w-3xl lg:ml-24">
          <strong>Produkt. Idee. Erfolg.</strong> Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in
          Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
        </p>
      </Section>

      <nav className="justify-around hidden px-20 my-8 text-lg font-semibold lg:flex">
        <Link href="/#crossflex">
          <a className="px-2">Crossflex</a>
        </Link>
        <Link href="/#purity">
          <a className="px-2">Purity</a>
        </Link>
        <Link href="/#cerastar">
          <a className="px-2 border-b-4 border-hdm">Cerastar</a>
        </Link>
        <Link href="/#massivholztueren">
          <a className="px-2">Massivholztüren</a>
        </Link>
        <Link href="/#moebelbauplatten">
          <a className="px-2">Möbelbauplatten</a>
        </Link>
        <Link href="/#dazu-zubehoer">
          <a className="px-2">Dazu Zubehör</a>
        </Link>
      </nav>
    </header>
  )
}

export default Header
