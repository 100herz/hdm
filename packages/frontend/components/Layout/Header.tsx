import NavLink from 'next/link'
import Image from 'next/image'
import { useRouter } from 'next/router'
import { Link } from 'react-scroll'
import Sticky from 'react-stickynode'
import styled from '@emotion/styled'

import { useWindowDimensions } from '@/hooks/use-window-dimensions'

export default function Header({ className = '' }: React.HTMLAttributes<HTMLElement>) {
  const router = useRouter()
  const { width = 0 } = useWindowDimensions()

  return (
    <header className={`relative overflow-x-hidden ${className}`}>
      <div className="justify-between mt-6 lg:container lg:flex">
        <h1 className="pt-8 mb-10 lg:mb-0" data-testid="main-logo">
          <NavLink href="/">
            <a className="block w-56 mx-auto" aria-label="HDM - Holz Design Moers GmbH">
              <Image src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" width={224} height={57} />
            </a>
          </NavLink>
        </h1>

        <Sticky enabled={width < 1024} innerZ="101">
          {router && router.pathname === '/' ? (
            <nav
              className="flex justify-between py-3 font-semibold bg-white sm:justify-around lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0"
              aria-label="Hauptnavigation auf der Startseite und in der mobilen Ansicht."
            >
              <Link
                to="produkte"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-120}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Produkte
              </Link>
              <Link
                to="unternehmen"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-60}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Unternehmen
              </Link>
              <Link
                to="kontakt"
                activeClass="active"
                spy={true}
                smooth={true}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Kontakt
              </Link>
            </nav>
          ) : (
            <nav
              className="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0"
              aria-label="Hauptnavigation nicht auf der Startseite aber in der mobilen Ansicht."
            >
              <NavLink href="/#produkte">
                <a className="px-4 py-2 my-2 hover:text-hdm">Produkte</a>
              </NavLink>
              <NavLink href="/#unternehmen">
                <a className="px-4 py-2 my-2 hover:text-hdm">Unternehmen</a>
              </NavLink>
              <NavLink href="/#kontakt">
                <a className="px-4 py-2 my-2 hover:text-hdm">Kontakt</a>
              </NavLink>
            </nav>
          )}
        </Sticky>
      </div>

      <Section className="container pt-12 pb-20 lg:flex lg:flex-row lg:justify-between lg:pt-32 lg:pb-24">
        <p className="text-xl leading-10 lg:text-3xl lg:max-w-3xl lg:ml-24">
          <strong>Produkt. Idee. Erfolg.</strong> Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in
          Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
        </p>
        {router && router.pathname === '/' ? (
          <Link to="purity" activeClass="active" spy={true} smooth={true} offset={-25} duration={700}>
            <figure id="landing-eco" className="hidden w-56 -mt-16 text-center group lg:block">
              <Image
                className="transition-all duration-500 transform -rotate-10 group-hover:rotate-0"
                src="/certificates/hdm-oekosiegel-header.png"
                alt="HDM - Verantwortung für Mensch und Natur"
                title="Mehr über die den BIO ... logischen Laminatboden erfahren"
                width={300}
                height={300}
              />
            </figure>
          </Link>
        ) : (
          <figure className="hidden w-56 -mt-16 text-center lg:block">
            <Image
              src="/certificates/hdm-oekosiegel-header.png"
              alt="HDM - Verantwortung für Mensch und Natur"
              title="Mehr über die den BIO ... logischen Laminatboden erfahren"
              width={300}
              height={300}
            />
          </figure>
        )}
      </Section>

      {router && router.pathname === '/' && (
        <Sticky enabled={width >= 1024} innerZ="101" activeClass="sticky-nav">
          <nav
            className="justify-around hidden h-24 px-20 py-4 text-lg font-semibold lg:flex"
            aria-label="Produktnavigation in der Desktopansicht."
          >
            <Link
              to="crossflex"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-120}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Dekorpaneele
            </Link>
            <Link
              to="purity"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-25}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Laminatb&ouml;den
            </Link>
            <Link
              to="cerastar"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Designb&ouml;den
            </Link>
            <Link
              to="massivholztueren"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Massivholzt&uuml;ren
            </Link>
            <Link
              to="shelves"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Möbelbauteile
            </Link>
            <Link
              to="strips"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Leisten
            </Link>
            <Link
              to="accessories"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              {!process.browser ? 'DAZU-Zubehör' : width < 1280 ? 'Zubehör' : 'DAZU-Zubehör'}
            </Link>
          </nav>
        </Sticky>
      )}
    </header>
  )
}

const Section = styled.section`
  position: relative;

  &::before {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: -2;
    background-color: #f7fafc;
    content: '';
  }

  &::after {
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
    width: calc(2.9333 * 90%);
    height: 90%;
    background-image: url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    opacity: 0.05;
    content: '';
  }

  @media (min-width: 1024px) {
    position: static;

    &::after {
      left: 6rem;
      width: calc(2.9333 * 100%);
      height: 100%;
    }
  }
`
