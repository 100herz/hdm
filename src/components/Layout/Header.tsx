import NavLink from "next/link"
import { useRouter } from "next/router"
import { Link } from "react-scroll"
import Sticky from "react-stickynode"
import styled from "styled-components"

import { useWindowDimensions } from "@/hooks/useWindowDimensions"

const Section = styled.section`
  position: relative;

  &::before {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: -2;
    background-color: #f7fafc;
    content: "";
  }

  &::after {
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
    width: calc(2.9333 * 90%);
    height: 90%;
    background-image: url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    opacity: 0.05;
    content: "";
  }

  @media (min-width: 768px) {
    position: static;

    &::after {
      left: 6rem;
      width: calc(2.9333 * 100%);
      height: 100%;
    }
  }
`

type Props = {
  className?: string
}

const Header: React.FC<Props> = ({ className = "" }: Props) => {
  const router = useRouter()
  const { width = 0 } = useWindowDimensions()

  return (
    <header className={`relative overflow-x-hidden ${className}`}>
      <div className="justify-between mt-6 lg:container lg:flex">
        <h1 className="pt-8 mb-10 lg:mb-0">
          <NavLink href="/">
            <a>
              <img className="w-56 mx-auto" src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" />
              <span className="sr-only">HDM - Holz Design Moers GmbH</span>
            </a>
          </NavLink>
        </h1>

        <Sticky enabled={width < 768} innerZ="101">
          {router && router.pathname === "/" ? (
            <nav className="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0">
              <Link
                to="produkte"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-120}
                duration={600}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Produkte
              </Link>
              <Link
                to="unternehmen"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-60}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Unternehmen
              </Link>
              <Link
                to="kontakt"
                activeClass="active"
                spy={true}
                smooth={true}
                duration={1200}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Kontakt
              </Link>
            </nav>
          ) : (
            <nav className="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0">
              <NavLink href="/#produkte">
                <a className="px-4 py-2 my-2 hover:text-hdm">Produkte</a>
              </NavLink>
              <NavLink href="/#unternehmen">
                <a className="px-4 py-2 my-2 hover:text-hdm">Unternehmen</a>
              </NavLink>
              <NavLink href="/#kontakt">
                <a className="px-4 py-2 my-2 hover:text-hdm">Kontakt</a>
              </NavLink>
            </nav>
          )}
        </Sticky>
      </div>

      <Section className="container pt-12 pb-20 lg:flex lg:flex-row lg:justify-between lg:pt-32 lg:pb-24">
        <p className="text-xl leading-10 lg:text-3xl lg:max-w-3xl lg:ml-24">
          <strong>Produkt. Idee. Erfolg.</strong> Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in
          Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
        </p>
        <Link to="purity" activeClass="active" spy={true} smooth={true} offset={-25} duration={700}>
          <a className="hidden -mt-16 lg:block hover:text-green-600">
            <figure className="text-center group">
              <img
                className="w-56 transition-all duration-500 transform -rotate-10 group-hover:rotate-0"
                src="/certificates/hdm-oekosiegel-header.png"
                alt="HDM - Verantwortung für Mensch und Natur"
                title="Mehr über die den BIO ... logischen Laminatboden erfahren"
              />
            </figure>
          </a>
        </Link>
      </Section>

      {router && router.pathname === "/" && (
        <Sticky enabled={width >= 768} innerZ="101" activeClass="sticky-nav">
          <nav className="justify-around hidden h-24 px-20 py-4 text-lg font-semibold lg:flex">
            <Link
              to="crossflex"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-120}
              duration={600}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Dekorpaneele
            </Link>
            <Link
              to="purity"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-25}
              duration={700}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Laminatb&ouml;den
            </Link>
            <Link
              to="cerastar"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={900}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Designb&ouml;den
            </Link>
            <Link
              to="massivholztueren"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Massivholzt&uuml;ren
            </Link>
            <Link
              to="accessories"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1100}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              {width >= 768 ? "Regalböden / Möbelbauplatten" : "Böden"}
            </Link>
            <Link
              to="accessories"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1100}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              DAZU<sup>&reg;</sup> Zubeh&ouml;r
            </Link>
          </nav>
        </Sticky>
      )}
    </header>
  )
}

export default Header
