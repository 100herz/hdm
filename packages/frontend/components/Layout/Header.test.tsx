import { render } from '@/test/test-utils'

import Header from './Header'

describe('Header component', () => {
  it('matches snapshot', () => {
    const { container, getByTestId } = render(<Header />)
    expect(getByTestId(/main-logo/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})

// TODO: Test the `links` and `utils`
