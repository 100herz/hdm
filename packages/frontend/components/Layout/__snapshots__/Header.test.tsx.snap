// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-0::before {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: -2;
  background-color: #f7fafc;
  content: '';
}

.emotion-0::after {
  position: absolute;
  top: 0;
  left: 0;
  z-index: -1;
  width: calc(2.9333 * 90%);
  height: 90%;
  background-image: url(/ci/hdm-bg.svg);
  background-repeat: no-repeat;
  opacity: 0.05;
  content: '';
}

@media (min-width: 1024px) {
  .emotion-0 {
    position: static;
  }

  .emotion-0::after {
    left: 6rem;
    width: calc(2.9333 * 100%);
    height: 100%;
  }
}

<header
  class="relative overflow-x-hidden "
>
  <div
    class="justify-between mt-6 lg:container lg:flex"
  >
    <h1
      class="pt-8 mb-10 lg:mb-0"
      data-testid="main-logo"
    >
      <a
        aria-label="HDM - Holz Design Moers GmbH"
        class="block w-56 mx-auto"
        href="/"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjI0IiBoZWlnaHQ9IjU3IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <img
            alt="Logo der HDM GmbH"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </a>
    </h1>
    <div
      class="sticky-outer-wrapper"
    >
      <div
        class="sticky-inner-wrapper "
        style="position: relative; top: 0px; z-index: 101;"
      >
        <nav
          aria-label="Hauptnavigation nicht auf der Startseite aber in der mobilen Ansicht."
          class="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0"
        >
          <a
            class="px-4 py-2 my-2 hover:text-hdm"
            href="/#produkte"
          >
            Produkte
          </a>
          <a
            class="px-4 py-2 my-2 hover:text-hdm"
            href="/#unternehmen"
          >
            Unternehmen
          </a>
          <a
            class="px-4 py-2 my-2 hover:text-hdm"
            href="/#kontakt"
          >
            Kontakt
          </a>
        </nav>
      </div>
    </div>
  </div>
  <section
    class="container pt-12 pb-20 lg:flex lg:flex-row lg:justify-between lg:pt-32 lg:pb-24 emotion-0 emotion-1"
  >
    <p
      class="text-xl leading-10 lg:text-3xl lg:max-w-3xl lg:ml-24"
    >
      <strong>
        Produkt. Idee. Erfolg.
      </strong>
       Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
    </p>
    <figure
      class="hidden w-56 -mt-16 text-center lg:block"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt="HDM - Verantwortung für Mensch und Natur"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          title="Mehr über die den BIO ... logischen Laminatboden erfahren"
        />
      </div>
    </figure>
  </section>
</header>
`;
